yt/utilities/linear_interpolators.py

```diff
@@ -1,11 +1,82 @@
+import abc
+
 import numpy as np
 
 import yt.utilities.lib.interpolators as lib
 from yt.funcs import mylog
 
 
-class UnilinearFieldInterpolator:
-    def __init__(self, table, boundaries, field_names, truncate=False):
+class _LinearInterpolator(abc.ABC):
+    _ndim: int
+    _dim_i_type = "int32"
+
+    def __init__(self, table, field_names, truncate=False, *, store_table=True):
+        if store_table:
+            self.table = table.astype("float64")
+        else:
+            self.table = None
+        self.table_shape = table.shape
+        self.truncate = truncate
+        self._field_names = field_names
+
+    def _raise_truncation_error(self, data_object):
+        mylog.error(
+            "Sorry, but your values are outside "
+            "the table!  Dunno what to do, so dying."
+        )
+        mylog.error("Error was in: %s", data_object)
+        raise ValueError
+
+    def _validate_table(self, table_override):
+        if table_override is None:
+            if self.table is None:
+                msg = (
+                    f"You must either store the table used when initializing a new "
+                    f"{type(self).__name__} (set `store_table=True`) or you must provide a `table` when "
+                    f"calling {type(self).__name__}"
+                )
+                raise ValueError(msg)
+            return self.table
+
+        if table_override.shape != self.table_shape:
+            msg = f"The table_override shape, {table_override.shape}, must match the base table shape, {self.table_shape}"
+            raise ValueError(msg)
+
+        return table_override.astype("float64")
```

Comment:
again, is the implicit copy intended ?

Reply:
same reason as before -- I just hadn't dug into the reasons for copying the data in the original code, so decided to just follow what the old code was doing.

Reply:
Ok, looking through the code again, I don't see a reason why we'd need a copy.  `table_override.astype("float64", copy=False)` should be totally fine here.